tools/explorer/pkg/workloads/reservation.go

```diff
@@ -58,10 +69,15 @@ func (a *API) validAddresses(ctx context.Context, db *mongo.Database, res *types
 		return err
 	}
 
-	validator := stellar.NewAddressValidator(config.Config.Network, config.Config.Asset)
-
 	for _, farm := range farms {
 		for _, a := range farm.WalletAddresses {
+			validator, err := stellar.NewAddressValidator(config.Config.Network, a.Asset)
```

Comment:
What is going to happen if someone has in its wallet addresses an asset not supported by the explorer? 
As example, now most of the farm have a fake TFChain asset code in their wallet addesses.

Reply:
`NewAddressValidator` has been slightly reworked, it now creates a wallet, and tries to load the given asset code. If this fails, an `ErrAssetCodeNotSupported` is returned by the wallet, which is then wrapped and propagated up. 2 lines below, this code, we check if the returned errors `Is` an instance of this error, and ignore this address should this be the case. So we only validate addresses which are associated with a known asset(code). This should be the same behavior as the old implementation afaik

Reply:
Alright perfect.